### ISSUE TITLE
Catch more exceptions in main()

### ISSUE THREAD

**Issue by elfring** [endless-sky/endless-sky#126]
I wonder why your function "[main](https://github.com/endless-sky/endless-sky/blob/9a7212716834fcbbd9acc9525bca0867b4004a4b/source/main.cpp#L51 "main() function")" does not contain [catch clauses for more exception types](http://stackoverflow.com/questions/368184/does-it-make-sense-to-catch-exceptions-in-the-main "Does it make sense to catch exceptions in the main(…)?") so far.

How do you think about recommendations by Matthew Wilson in [an article](http://accu.org/index.php/journals/1706 "Quality Matters #6: Exceptions for Practically-Unrecoverable Conditions")?

Would you like to adjust the implementation if you consider [effects for uncaught/unhandled exceptions](http://www.informit.com/guides/content.aspx?g=cplusplus&seqNum=108 "Uncaught exceptions") like they are described by Danny Kalev?

**Comment by endless-sky** [comment]
All standard C++ exceptions are subclasses of runtime_error, including all the exceptions that my code throws. So, catching runtime_error is sufficient in this case.

**Comment by elfring** [comment]
* How do you think about to take extra care for a C++ exception like "[std::bad_alloc](http://en.cppreference.com/w/cpp/memory/new/bad_alloc "Allocation failure")"?
* Would you like to add [a catch-all clause](http://en.cppreference.com/w/cpp/language/try_catch "try-block") there?

**Comment by endless-sky** [comment]
If an alloc error or some other unknown error occurs, terminating the program immediately is probably a better solution than attempting to display an SDL error message, etc., because something is seriously wrong with the program or the OS. So, I'm going to leave this as-is.

**Comment by cmyster** [comment]
Hi, I totally agree with exit immediately but since I am hitting ^ often, is there any way to print something anyway? maybe a --debug or the likes.
GDB isn't helpful to me here (maybe someone can make use of this):
http://paste.openstack.org/show/480396

**Comment by zwparchman** [comment]
What else would you want? Gdb is letting you know that of the 8 threads only one isn't waiting on a condition variable and thread 1 failed to allocate memory for a  vector emplace_back operation.

Your problem however may be that there isn't enough memory on the system. If this is a Linux based system bad_alloc is normally only thrown after attempting to allocate more memory than the system has (including swap) in one call. The failure mode for "more memory in use than the system has" is the kernel picks a process to kill. Look into the Out Of Memory Killer if you want more information. If this problem persists you may want to add a larger swap partition; although, looking at where that problem is (reallocating a likely to be short list of pointers) there is likely a problem elsewhere that logging this exception won't fix.

**Comment by endless-sky** [comment]
This particular out of memory error is actually, I think, due to an infinite loop if you click on a system with no known path to it. That's a bug that I introduced in commit 6abfa14 and fixed just now in 74f1420.

**Comment by cmyster** [comment]
@zwparchman Thats why I was confused, it had more then enough free RAM at the time.
@endless-sky Many thanks, and a great game!
